### PR TITLE
MF - 1590 - Fix fetching list of users with a zero limit

### LINKED ISSUE #1590: Fetching list of users with a zero limit
**BUG REPORT**

1. What were you trying to achieve?
I am trying to query the list of users with a zero limit

2. What are the expected results?
I should receive `400 Bad Request with error invalid limit size`

```sh
HTTP/1.1 400 Bad Request
Server: nginx/1.20.0
Date: Wed, 06 Apr 2022 14:34:10 GMT
Content-Type: text/plain; charset=utf-8
Content-Length: 31
Connection: keep-alive

{"error":"invalid limit size"}

```
3. What are the received results?
I have received 
```sh
HTTP/1.1 200 OK
Server: nginx/1.20.0
Date: Wed, 06 Apr 2022 14:34:04 GMT
Content-Type: application/json
Content-Length: 45
Connection: keep-alive
Strict-Transport-Security: max-age=63072000; includeSubdomains
X-Frame-Options: DENY
X-Content-Type-Options: nosniff
Access-Control-Allow-Origin: *
Access-Control-Allow-Methods: *
Access-Control-Allow-Headers: *

{"total":13,"offset":0,"limit":0,"users":[]}

```
4. What are the steps to reproduce the issue?
- Start mainflux services i.e `docker-compose -f docker/docker-compose up -d`
- Try and query users endpoint with zero limit i.e `curl -s -S -i -X GET -H "Authorization: Bearer $USERTOKEN" http://localhost/users\?offset\=0\&limit\=0\&name\=`

5. In what environment did you encounter the issue?
- base environment

6. Additional information you deem important:
N/A

### PR DESCRIPTION
**Pull request by rodneyosodo**
Signed-off-by: 0x6f736f646f <blackd0t@protonmail.com>

### What does this do?
It adds a max limit size and a min limit size when fetching the list of users. It also adds max email size

### Which issue(s) does this PR fix/relate to?
Resolves #1590

### List any changes that modify/break current functionality
Throws and `invalid limit size` when using a limit size of 0 or greater than 100

### Have you included tests for your changes?
No

### Did you document any new/modified functionality?
No

### Notes
N/A

### CODE REVIEW

**Comment by manuio** [comment]
@0x6f736f646f Can you update your branch please? Aslo, can you check here: https://github.com/mainflux/mainflux/blob/195b78303f0a13ac4eb9ad639b2329b3773b807c/users/service.go#L80 You can probably use PageMetadata as it's done here for things list: https://github.com/mainflux/mainflux/blob/master/things/api/things/http/requests.go#L255

**Comment by rodneyosodo** [comment]
The changes requested are for #1589. Let me implement them there.

**Review by manuio — DISMISSED** [review]
LGTM

File: users/api/requests.go [hunk 1]
```diff
@@ -49,6 +54,14 @@ func (req listUsersReq) validate() error {
 	if req.token == "" {
 		return apiutil.ErrBearerToken
 	}
+	if req.limit > maxLimitSize || req.limit < 1 {
```
Comment (dborovcanin) [inline users/api/requests.go:58]: Please ad an empty line to be consistent with the rest of the code.

**Review by manuio — APPROVED** [review]
LGTM

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**